### PR TITLE
Run sos test inside docker and fix dotnet code viewer

### PR DESCRIPTION
**Pull request by vtikoo**
Signed-off-by: Vikas Tikoo <vitikoo@microsoft.com>

### CODE REVIEW

File: tests/dotnet-sos/Makefile [hunk 1]
```diff
@@ -25,21 +25,22 @@ ext2fs: appdir
 oelldb:
 	git clone --branch working https://github.com/vtikoo/oelldb
 
-dotnet-sos:
-	sudo apt install -y dotnet-sdk-5.0 && \
-	dotnet tool install -g dotnet-sos \
-	&& $(HOME)/.dotnet/tools/dotnet-sos install || true
-
 clean:
 	sudo rm -fr appdir ext2fs rootfs roothash oelldb stdouterr.txt /tmp/myst*
```
Comment (mikbras) [inline tests/dotnet-sos/Makefile:29]: This is fails on my box when trying to remove /tmp/myst*. It says it is busy. It is probably okay to leave /tmp/myst* files in place. They are removed by the pipeline first when it starts to run.

**Review by mikbras — CHANGES_REQUESTED** [review]
Looks good. The only problem I see is trying to remove /tmp/myst*. I ran it on my box and it seems good to me otherwise.

File: .azure_pipelines/ci-pipeline-code-coverage-nightly.yml [hunk 2]
```diff
@@ -29,7 +29,7 @@ jobs:
         curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
         sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
         sudo apt-get update
-        sudo apt-get install build-essential lcov python3-setuptools python3-pip llvm-7 libmbedtls-dev docker-ce lldb-10 -y
+        sudo apt-get install build-essential lcov python3-setuptools python3-pip llvm-7 libmbedtls-dev docker-ce -y
```
Comment (anakrish) [inline .azure_pipelines/ci-pipeline-code-coverage-nightly.yml:32]: Is the removal of lldb-10 intended?

Comment (vtikoo) [reply .azure_pipelines/ci-pipeline-code-coverage-nightly.yml:32]: Yes, its being removed from the pipeline VM setup script. The PR moves the SOS test to be run inside a docker container. The docker image has lldb-10 installed.

File: tests/dotnet-sos/Makefile (hunk 1 above)
Comment (vtikoo) [reply tests/dotnet-sos/Makefile]: Fixed by not removing /tmp/myst* as part  of `clean` target.

**Review by anakrish — APPROVED** [review]
Looks good from my end.

**Review by jxyang — APPROVED** [review]

**Review by mikbras — APPROVED** [review]

**Pull request merged**